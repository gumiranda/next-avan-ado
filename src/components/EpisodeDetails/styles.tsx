import styled from 'styled-components';

export const Wrapper = styled.div`
  max-width: 72rem;
  padding: 5rem 4rem;
  margin: 0 auto;
  header {
    padding-bottom: 1.6rem;
    border-bottom: 1px solid var(--gray-100);
    h1 {
      margin-top: 3.2rem;
      margin-bottom: 1.8rem;
    }
    span {
      display: inline-block;
      font-size: 1.4rem;
      & + span {
        margin-left: 1.6rem;
        padding-left: 1.6rem;
        position: relative;
        &::before {
          content: '';
          width: 4px;
          height: 4px;
          border-radius: 2px;
          background: #000;
          position: relative;
          left: 0;
          top: 50%;
          transform: translate(-50%, -50%);
        }
      }
    }
  }
`;
export const ThumbnailContainer = styled.div`
  position: relative;
  img {
    border-radius: 1.6rem;
  }
  button {
    width: 4.5rem;
    height: 4.5rem;
    border-radius: 1.5rem;
    border: 0;
    position: absolute;
    z-index: 5;
    font-size: 0;
    transition: filter 0.2s;
    &:hover {
      filter: brightness(0.9);
    }
    &:first-child {
      left: 0;
      top: 50%;
      background: var(--purple-500);
      transform: translate(-50%, -50%);
    }
    &:last-child {
      right: 0;
      top: 50%;
      background: var(--green-500);
      transform: translate(50%, -50%);
    }
  }
`;
export const DescriptionContainer = styled.div`
  margin-top: 3.2rem;
  line-height: 2rem;
  color: var(--gray-800);
  p {
    font-size: 1.6rem;
    margin: 1.5rem 0;
  }
`;
