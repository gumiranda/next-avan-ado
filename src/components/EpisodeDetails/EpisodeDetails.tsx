import * as S from './styles';
import Image from 'next/image';
import Link from 'next/link';
import { PlayerContext } from '@/domain/contexts';
import { useContext } from 'react';

interface IProps {
  episode: any;
}
const EpisodeDetails = ({ episode = {}, ...props }: IProps) => {
  const { play } = useContext(PlayerContext);

  return (
    <S.Wrapper>
      <S.ThumbnailContainer>
        <Link href="\">
          <button type="button">
            <img src="/arrow-left.svg" alt="Voltar" />
          </button>
        </Link>
        <Image width={700} height={220} src={episode.thumbnail} objectFit="cover" />
        <button type="button" onClick={() => play(episode)}>
          <img src="/play.svg" alt="Tocar episÃ³dio" />
        </button>
      </S.ThumbnailContainer>
      <header>
        <h1>{episode.title}</h1>
        <span>{episode.members}</span>
        <span>{episode.publishedAt}</span>
        <span>{episode.durationAsString}</span>
      </header>
      <S.DescriptionContainer dangerouslySetInnerHTML={{ __html: episode.description }} />
    </S.Wrapper>
  );
};
export default EpisodeDetails;
