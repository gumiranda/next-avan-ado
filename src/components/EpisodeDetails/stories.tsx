import { Story, Meta } from '@storybook/react/types-6-0';

import EpisodeDetails from './EpisodeDetails';

export default {
  title: 'EpisodeDetails',
  component: EpisodeDetails
} as Meta;

export const Basic: Story = (args) => <EpisodeDetails episode={{}} {...args} />;
