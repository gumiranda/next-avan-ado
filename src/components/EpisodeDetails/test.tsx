import { render, screen } from '@testing-library/react';

import EpisodeDetails from './EpisodeDetails';

//It works for me calling jest.mock before the Component Test Case like you mentioned:

jest.mock('next/config', () => () => ({
  images: {
    domains: ['storage.googleapis.com']
  }
}));
describe('<EpisodeDetails />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <EpisodeDetails
        episode={{
          title: 'test',
          thumbnail: '/logo.svg'
        }}
      />
    );
    expect(screen.getByRole('heading', { name: /test/i })).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
