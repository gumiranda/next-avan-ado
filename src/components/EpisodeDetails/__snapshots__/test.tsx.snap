// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EpisodeDetails /> should render the heading 1`] = `
.c0 {
  max-width: 72rem;
  padding: 5rem 4rem;
  margin: 0 auto;
}

.c0 header {
  padding-bottom: 1.6rem;
  border-bottom: 1px solid var(--gray-100);
}

.c0 header h1 {
  margin-top: 3.2rem;
  margin-bottom: 1.8rem;
}

.c0 header span {
  display: inline-block;
  font-size: 1.4rem;
}

.c0 header span + span {
  margin-left: 1.6rem;
  padding-left: 1.6rem;
  position: relative;
}

.c0 header span + span::before {
  content: '';
  width: 4px;
  height: 4px;
  border-radius: 2px;
  background: #000;
  position: relative;
  left: 0;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1 {
  position: relative;
}

.c1 img {
  border-radius: 1.6rem;
}

.c1 button {
  width: 4.5rem;
  height: 4.5rem;
  border-radius: 1.5rem;
  border: 0;
  position: absolute;
  z-index: 5;
  font-size: 0;
  -webkit-transition: filter 0.2s;
  transition: filter 0.2s;
}

.c1 button:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.c1 button:first-child {
  left: 0;
  top: 50%;
  background: var(--purple-500);
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1 button:last-child {
  right: 0;
  top: 50%;
  background: var(--green-500);
  -webkit-transform: translate(50%,-50%);
  -ms-transform: translate(50%,-50%);
  transform: translate(50%,-50%);
}

.c2 {
  margin-top: 3.2rem;
  line-height: 2rem;
  color: var(--gray-800);
}

.c2 p {
  font-size: 1.6rem;
  margin: 1.5rem 0;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <button
      type="button"
    >
      <img
        alt="Voltar"
        src="/arrow-left.svg"
      />
    </button>
    <div
      style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="box-sizing: border-box; display: block; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNzAwIiBoZWlnaHQ9IjIyMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
        />
      </div>
      <noscript />
      <img
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
      />
    </div>
    <button
      type="button"
    >
      <img
        alt="Tocar episÃ³dio"
        src="/play.svg"
      />
    </button>
  </div>
  <header>
    <h1>
      test
    </h1>
    <span />
    <span />
    <span />
  </header>
  <div
    class="c2"
  />
</div>
`;
